src/app/project_runner.go

```diff
@@ -257,19 +257,30 @@ func (p *ProjectRunner) initProcessLog(name string) {
 	p.processLogs[name] = pclog.NewLogBuffer(p.project.LogLength)
 }
 
-func (p *ProjectRunner) GetProcessState(name string) (*types.ProcessState, error) {
+func (p *ProjectRunner) GetProcessState(name string) (*types.ProcessState, bool, error) {
 	proc := p.getRunningProcess(name)
 	if proc != nil {
-		return proc.getState(), nil
+		return proc.getState(), false, nil
 	} else {
 		p.statesMutex.Lock()
 		defer p.statesMutex.Unlock()
+		p.procConfMutex.Lock()
+		defer p.procConfMutex.Unlock()
+		// config and state stored and modified not under same mutex,
+		// so state may not (yet, briefly) exist, but config does
+		config, ok := p.project.Processes[name]
+		if !ok {
+			return nil, true, fmt.Errorf("no such process in project: %s", name)
+		}
 		state, ok := p.processStates[name]
 		if !ok {
+			if config.Disabled {
```

Comment:
so we may not have state, because it is not yet here because not under same mutex, but config is disabled, so we need to exit early